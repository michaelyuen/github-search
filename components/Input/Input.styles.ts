import styled from "styled-components";

export const StyledInput = styled.input`
  padding: 0.5em;
`;
