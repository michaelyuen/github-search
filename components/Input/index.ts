export * from "./Input";
export * from "./styles";
