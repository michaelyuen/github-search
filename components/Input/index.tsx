export const Input: React.FC = () => {
  return <input type="text" />;
};
