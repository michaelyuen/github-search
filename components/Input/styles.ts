import styled from "styled-components";

export const StyledInput = styled.input`
  border: 1px solid var(--border-color);
  border-radius: var(--border-radius);
  height: 50px;
  padding: 0 1em;
`;
