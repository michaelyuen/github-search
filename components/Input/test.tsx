import { render, screen } from "@testing-library/react";
import { axe } from "jest-axe";
import { Input } from "./Input";

describe("Input", () => {
  test("renders", () => {
    render(<Input aria-label="mock aria label" />);
    expect(screen.getByLabelText("mock aria label")).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const { container } = render(<Input aria-label="mock aria label" />);
    expect(container).toMatchSnapshot();
  });

  test("has no a11y violations", async () => {
    const { container } = render(<Input aria-label="mock aria label" />);
    expect(await axe(container)).toHaveNoViolations();
  });
});
