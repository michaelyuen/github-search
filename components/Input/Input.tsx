import { HTMLInputTypeAttribute } from "react";
import { StyledInput } from "./styles";

export interface InputProps {
  disabled?: boolean;
  name?: string;
  isInvalid?: boolean;
  onBlur?(event: React.FormEvent<HTMLInputElement>): void;
  onChange?(event: React.FormEvent<HTMLInputElement>): void;
  type?: HTMLInputTypeAttribute;
  value?: string;
}

export const Input: React.FC<InputProps> = ({ type = "text", ...rest }) => {
  return <StyledInput type={type} {...rest} />;
};
