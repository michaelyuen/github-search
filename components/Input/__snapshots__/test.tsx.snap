// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input matches snapshot 1`] = `
<div>
  <input
    aria-label="mock aria label"
    autocapitalize="off"
    class="sc-bcXHqe iggXoH"
    spellcheck="false"
    type="text"
  />
</div>
`;
